PWGHF/Core/HfMlResponseD0ToKPi.h

```diff
@@ -0,0 +1,264 @@
+// or submit itself to any jurisdiction.
+
+/// \file HfMlResponseD0ToKPi.h
+/// \brief Class to compute the ML response for D0 → K∓ π± analysis selections
+/// \author Alexandre Bigot <alexandre.bigot@cern.ch>, IPHC Strasbourg
+/// \author Andrea Tavira García <tavira-garcia@ijclab.in2p3.fr>, IJCLab Orsay
+
+#ifndef PWGHF_CORE_HFMLRESPONSED0TOKPI_H_
+#define PWGHF_CORE_HFMLRESPONSED0TOKPI_H_
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "PWGHF/Core/HfHelper.h"
+#include "PWGHF/Core/HfMlResponse.h"
+
+// Fill the map of available input features
+// the key is the feature's name (std::string)
+// the value is the corresponding value in EnumInputFeatures
+#define FILL_MAP_D0(FEATURE)                                        \
+  {                                                                 \
+#FEATURE, static_cast < uint8_t>(InputFeaturesD0ToKPi::FEATURE) \
+  }
+
+// Check if the index of mCachedIndices (index associated to a FEATURE)
+// matches the entry in EnumInputFeatures associated to this FEATURE
+// if so, the inputFeatures vector is filled with the FEATURE's value
+// by calling the corresponding GETTER from OBJECT
+#define CHECK_AND_FILL_VEC_D0_FULL(OBJECT, FEATURE, GETTER)   \
+  case static_cast<uint8_t>(InputFeaturesD0ToKPi::FEATURE): { \
+    inputFeatures.emplace_back(OBJECT.GETTER());              \
+    break;                                                    \
+  }
+
+// Specific case of CHECK_AND_FILL_VEC_D0_FULL(OBJECT, FEATURE, GETTER)
+// where OBJECT is named candidate and FEATURE = GETTER
+#define CHECK_AND_FILL_VEC_D0(GETTER)                        \
+  case static_cast<uint8_t>(InputFeaturesD0ToKPi::GETTER): { \
+    inputFeatures.emplace_back(candidate.GETTER());          \
+    break;                                                   \
+  }
+
+// Variation of CHECK_AND_FILL_VEC_D0_FULL(OBJECT, FEATURE, GETTER)
+// where GETTER is a method of hfHelper
+#define CHECK_AND_FILL_VEC_D0_HFHELPER(OBJECT, FEATURE, GETTER) \
+  case static_cast<uint8_t>(InputFeaturesD0ToKPi::FEATURE): {   \
+    inputFeatures.emplace_back(hfHelper.GETTER(OBJECT));        \
+    break;                                                      \
+  }
+
+namespace o2::analysis
+{
+enum class InputFeaturesD0ToKPi : uint8_t {
+  collisionId = 0,
+  posX,
+  posY,
+  posZ,
+  xSecondaryVertex,
+  ySecondaryVertex,
+  zSecondaryVertex,
+  errorDecayLength,
+  errorDecayLengthXY,
+  chi2PCA,
+  rSecondaryVertex,
+  decayLength,
+  decayLengthXY,
+  decayLengthNormalised,
+  decayLengthXYNormalised,
+  impactParameterNormalised0,
+  ptProng0,
+  pt2Prong0,
+  ptProng1,
+  pt2Prong1,
+  pxProng0,
+  pyProng0,
+  pzProng0,
+  pxProng1,
+  pyProng1,
+  pzProng1,
+  impactParameter0,
+  impactParameter1,
+  errorImpactParameter0,
+  errorImpactParameter1,
+  nSigTpcPi0,
+  nSigTpcKa0,
+  nSigTofPi0,
+  nSigTofKa0,
+  nSigTpcPi1,
+  nSigTpcKa1,
+  nSigTofPi1,
+  nSigTofKa1,
```

Comment:
In addition to the single Nsigma TPC and TOF variables, I would also add the combined one (see e.g. https://github.com/AliceO2Group/O2Physics/blob/1204d8f12614b5601381c70138bcdee4224ccf5b/PWGHF/Core/HfMlResponseDplusToPiKPi.h#L78-L83)